x-pack/plugins/rollup/server/plugin.ts

```diff
@@ -99,9 +102,10 @@ export class RollupPlugin implements Plugin<void, void, any, any> {
         this.logger.warn(`Registering Rollup collector failed: ${e}`);
       }
     }
-
-    if (indexManagement && indexManagement.indexDataEnricher) {
-      indexManagement.indexDataEnricher.add(rollupDataEnricher);
+    if (this.config.ui.enabled) {
```

Comment:
Would we still proceed with some of the other setup functionality if the UI is disabled? 🤔 

I see we have similar logic to this in other plugins so maybe it's OK.

Reply:
That is a great point actually, I will check other functionality and disable if needed.

Reply:
Added more changes in https://github.com/elastic/kibana/pull/167295/commits/6ba29ff88254e81383256f27a4f9a9a510e987c2 and updated the PR description